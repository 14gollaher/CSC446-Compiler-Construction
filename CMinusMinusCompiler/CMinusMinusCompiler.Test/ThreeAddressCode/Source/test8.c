const g = 33;
int main()
{
  int a,b,c,d,e,f;

  a=5;
  b=10;
  d= g + 20;
  c = g + (-a);
  return c;
}
