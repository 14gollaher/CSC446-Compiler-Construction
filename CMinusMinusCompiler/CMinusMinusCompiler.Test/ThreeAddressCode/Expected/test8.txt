PROC main
	_BP-14 = 5
	_BP-2 = _BP-14
	_BP-16 = 10
	_BP-4 = _BP-16
	_BP-18 = 33
	_BP-22 = 20
	_BP-20 = _BP-18 + _BP-22
	_BP-8 = _BP-20
	_BP-24 = 33
	_BP-26 = _BP-24 + -_BP-2
	_BP-6 = _BP-26
	_AX = _BP-6
ENDP main

START PROC main

